docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md

```diff
@@ -0,0 +1,64 @@
+---
+id: best-practices
+title: Best practices
+description: An index of best practices for plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - best practices
+sidebar_position: 5
+---
+
+This document summarizes best practices for Grafana plugins. This live document is updated as new best practices are identified.
+
+## General
+
+- Choose the right plugin type for your use case: panel, data-source or app. See [Plugin types](/introduction/plugin-types-usage) for more information.
+- Verify that your data-source or app plugin can be provisioned. See [Provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
```

Comment:
Is there something particular that a plugin dev needs to do to ensure provisioning works as expected?

Reply:
Difficult to say, we have found bugs when testing plugins when a provisioned data-source does not render properly properly in the config editor for example or when upgrading a plugin, when a provisioned data source for one version does not work in the next. Not sure if we can be more specific here. I just wanted to point out that provisioning is a thing and plugin devs should be aware of that.

Reply:
Oh interesting. By all means keep it - would be great if we could narrow down the cause/resolution though